### PR TITLE
Added Catalog Images(snapshots)

### LINKED ISSUE #1494: Add Images(Snapshots) in Catalog
https://layer5io.slack.com/archives/CFGG6U10E/p1699635627972819

### PR DESCRIPTION
**Pull request by coder12git**
**Description**

This PR fixes #1494

**Notes for Reviewers**

**[Signed commits](https://docs.meshery.io/project/contributing#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Review by MUzairS15 — COMMENTED** [review]
Don't edit the catalog manually.
Also can you share the screenshot of the page by explicity setting image attribute in of the catalog to `https://github.com/layer5labs/meshery-extensions-packages/blob/master/action-assets/design-assets/2512c0cb-385d-40de-b1f5-bc4a67131538.png` (Be sure to undo your changes)

**Comment by coder12git** [comment]
> Don't edit the catalog manually. Also can you share the screenshot of the page by explicity setting image attribute in of the catalog to `https://github.com/layer5labs/meshery-extensions-packages/blob/master/action-assets/design-assets/2512c0cb-385d-40de-b1f5-bc4a67131538.png` (Be sure to undo your changes)

Do you want this image for every catalog? What i understand was that everyone will have a different image, that's why i what i did was i added a simple check statement in main file to just check if we have image attribute in remote url, then we are going to show that image else for all other catalog's we will have default service-mesh image, that's why i removed that image attribute from those catalogs md files, which have these default image.

**Comment by coder12git** [comment]
> Don't edit the catalog manually. Also can you share the screenshot of the page by explicity setting image attribute in of the catalog to `https://github.com/layer5labs/meshery-extensions-packages/blob/master/action-assets/design-assets/2512c0cb-385d-40de-b1f5-bc4a67131538.png` (Be sure to undo your changes)

Sure, will do the changes. Btw this link that you provided here should be this-`https://raw.githubusercontent.com/layer5labs/meshery-extensions-packages/master/action-assets/design-assets/2512c0cb-385d-40de-b1f5-bc4a67131538.png`

**Comment by MUzairS15** [comment]
> > Don't edit the catalog manually. Also can you share the screenshot of the page by explicity setting image attribute in of the catalog to `https://github.com/layer5labs/meshery-extensions-packages/blob/master/action-assets/design-assets/2512c0cb-385d-40de-b1f5-bc4a67131538.png` (Be sure to undo your changes)
> 
> Sure, will do the changes. Btw this link that you provided here should be this-`https://raw.githubusercontent.com/layer5labs/meshery-extensions-packages/master/action-assets/design-assets/2512c0cb-385d-40de-b1f5-bc4a67131538.png`

Oh yes, I directly pasted the GitHub link and forgot to change it to the raw URL.

**Comment by coder12git** [comment]
@MUzairS15 done the changes, also i have fixed the formatting issue in two `.md ` files of catalog, which were giving build errors.
Screenshot 2023-11-12 004832

Screenshot 2023-11-12 010140

**Comment by abhijeetgauravm** [comment]
Hey @coder12git  Let's discuss this on website call tomorrow at 6:30 PM IST / 7 AM Central time. Please add this as an agenda item in the [meeting minutes](https://docs.google.com/document/d/1XczAHXVe2FIWPqiF57ospJ43zw5cZQ7ui8mn39v5EvA/edit#heading=h.lohhtewfwima).

**Comment by MUzairS15** [comment]
@coder12git Great.
Can you also add the capability to render the default image if the src is not found? i.e. if the src attribute points to an invalid URL then render the default icon?

**Comment by MUzairS15** [comment]
@coder12git Try this once `<img src="......" onerror="this.onerror=null; this.src='/assets/.....png'"`

To test, set an invalid URL for src.

**Comment by MUzairS15** [comment]
Oh, you already did. Perfect.

File: _includes/coming-soon.html [hunk 1]
```diff
@@ -3,7 +3,7 @@
             <div class="front">
                 <h4 class="overlay-text">{{pattern.name}}</h4>
                 <h5>Coming Soon...</h5>
-                <img class="pattern-image-overlay" src="{{pattern.image}}" />
+                <img class="pattern-image-overlay" src="https://raw.githubusercontent.com/layer5labs/meshery-extensions-packages/master/action-assets/design-assets/{{pttrn.patternId}}.png" onerror="this.src='/assets/images/patterns/service-mesh.svg'" />
```
Comment (MUzairS15) [inline _includes/coming-soon.html:6]:
```suggestion
                <img class="pattern-image-overlay" src="https://raw.githubusercontent.com/layer5labs/meshery-extensions-packages/master/action-assets/design-assets/{{pattern.patternId}}.png" onerror="this.src='/assets/images/patterns/service-mesh.svg'" />
```

**Review by MUzairS15 — APPROVED** [review]

**Pull request merged**